components/object_store/src/prefix.rs

```diff
@@ -130,11 +193,171 @@ impl ObjectStore for StoreWithPrefix {
+            Err(Error::NotImplemented)
+        }
+
+        async fn get_range(&self, location: &Path, _range: Range<usize>) -> Result<Bytes> {
+            self.prefix_checker.check(location);
+            Ok(self.content.clone())
+        }
+
+        async fn head(&self, location: &Path) -> Result<ObjectMeta> {
+            self.prefix_checker.check(location);
+
+            Ok(ObjectMeta {
+                location: location.clone(),
+                last_modified: DateTime::<Utc>::default(),
+                size: 0,
+            })
+        }
+
+        async fn delete(&self, location: &Path) -> Result<()> {
+            self.prefix_checker.check(location);
+
+            Err(Error::NotImplemented)
+        }
+
+        async fn list(&self, prefix: Option<&Path>) -> Result<BoxStream<'_, Result<ObjectMeta>>> {
+            if let Some(loc) = prefix {
+                self.prefix_checker.check(loc);
+            }
+            let prefix = prefix.map(|v| v.to_string()).unwrap_or_default();
+            let mut objects = Vec::with_capacity(self.file_num);
+            for file_idx in 0..self.file_num {
+                let raw_filepath = format!("{}/{}", prefix, file_idx);
+                let filepath = Path::from(raw_filepath);
+                let object = ObjectMeta {
+                    location: filepath,
+                    last_modified: DateTime::<Utc>::default(),
+                    size: 0,
+                };
+                objects.push(Ok(object));
+            }
+
+            Ok(stream::iter(objects).boxed())
+        }
+
+        async fn list_with_delimiter(&self, prefix: Option<&Path>) -> Result<ListResult> {
+            if let Some(loc) = prefix {
+                self.prefix_checker.check(loc);
+            }
+            Err(Error::NotImplemented)
+        }
+
+        async fn copy(&self, from: &Path, to: &Path) -> Result<()> {
+            self.prefix_checker.check(from);
+            self.prefix_checker.check(to);
+            Err(Error::NotImplemented)
+        }
+
+        async fn copy_if_not_exists(&self, from: &Path, to: &Path) -> Result<()> {
+            self.prefix_checker.check(from);
+            self.prefix_checker.check(to);
+            Err(Error::NotImplemented)
+        }
+    }
+
+    #[tokio::test]
+    async fn test_with_mock_store() {
+        let test_prefix = "test";
+        let mock_store = MockObjectStore {
+            file_num: 3,
+            content: Bytes::from_static(b"00000"),
+            prefix_checker: PathPrefixChecker {
+                prefix: test_prefix.to_string(),
+            },
+        };
+        let mock_store = Arc::new(mock_store);
+        let prefix_store = StoreWithPrefix::new(test_prefix.to_string(), mock_store).unwrap();
+
+        let test_filepath = Path::from("0/100");
+
+        let _ = prefix_store
+            .put(&test_filepath, Bytes::from_static(b"1111"))
+            .await;
+
+        let _ = prefix_store.get(&test_filepath).await;
```

Comment:
If you ignore those result, how can you ensure expected behavior?

Reply:
I have added some comments to explain this.